src/main/java/com/bankingservice/banking/services/AccountService.java

```diff
@@ -38,34 +31,46 @@ public class AccountService {
+            RegisterUserModel registerUserModel = accountServiceHelper.saveRegisterModel(entity);
+            RegisterUserResponseDTO registerUserResponseDTO = accountServiceHelper.makeRegisterUserResponseDTO(registerUserModel);
             return registerUserResponseDTO;
-        } catch (Exception e) {
-            logger.error("[insertDetailsForRegistration] details insertion failed in mysql for request: {}", registerRequestDTO, e);
-            throw new InsertionFailedException(ErrorCode.USER_REGISTRATION_FAILED, ErrorCode.USER_REGISTRATION_FAILED.getErrorMessage(), ErrorCode.USER_REGISTRATION_FAILED.getDisplayMessage());
+        } catch (DataIntegrityViolationException e) {
+            logger.error("[insertDetailsForRegistration] details insertion failed in db for request: {}", registerRequestDTO, e);
+            throw new InsertionFailedException(ErrorCode.USER_REGISTRATION_FAILED,
+                    ErrorCode.USER_REGISTRATION_FAILED.getErrorMessage(),
+                    ErrorCode.USER_REGISTRATION_FAILED.getDisplayMessage());
         }
     }
 
 
-    /**
-     * insert details for user on boarding
-     *
-     * @param onBoardRequestDTO
-     * @return onBoardResponseDTO
-     */
-    public OnBoardResponseDTO insertDetailsForOnBoarding(OnBoardRequestDTO onBoardRequestDTO) throws UserIdNotFoundException {
-        try {
-            logger.info("[insertDetailsForOnBoarding] On boarding a user: {}", onBoardRequestDTO);
-            UserOnBoardModel user = userOnBoardRepository.save(accountServiceHelper.makeOnBoardingEntity(onBoardRequestDTO));
-            OnBoardResponseDTO onBoardResponseDTO = accountServiceHelper.makeOnBoardingResponseDTO(user);
-            return onBoardResponseDTO;
-        } catch (UserIdNotFoundException e) {
-            logger.error("[insertDetailsForOnBoarding] details insertion failed in mysql for request: {}", onBoardRequestDTO, e);
-            throw new UserIdNotFoundException(ErrorCode.USER_ONBOARD_FAILED, ErrorCode.USER_ONBOARD_FAILED.getErrorMessage(), ErrorCode.USER_ONBOARD_FAILED.getDisplayMessage());
+        /**
+         * insert details for user on boarding
+         *
+         * @param onBoardRequestDTO
+         * @return onBoardResponseDTO
+         */
+        public OnBoardResponseDTO insertDetailsForOnBoarding (OnBoardRequestDTO onBoardRequestDTO) throws
+                UserIdNotFoundException, InsertionFailedException {
+            try {
+                logger.info("[insertDetailsForOnBoarding] On boarding a user: {}", onBoardRequestDTO);
+                UserOnBoardModel user = accountServiceHelper.makeOnBoardingEntity(onBoardRequestDTO);
+                UserOnBoardModel userOnBoardModel = accountServiceHelper.saveOnBoardModel(user);
+                OnBoardResponseDTO onBoardResponseDTO = accountServiceHelper.makeOnBoardingResponseDTO(userOnBoardModel);
+                return onBoardResponseDTO;
+            } catch (UserIdNotFoundException e) {
```

Comment:
need not require this catch as, this thrown exception can be handled by ApplicationExceptionHandler